gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/artifacts/DependencyDiscovery.java

```diff
@@ -40,17 +41,20 @@ public static ArtifactView getFilteredArtifact(Project project, Configuration co
     }
 
     public static Configuration copyConfiguration(Project project, String configurationName, String name) {
```

Comment:
I haven't fully figured out why but the `pdepsConfigProvider` get resolved twice for some projects and thus we need to make sure that we don't crash on subsequent calls.

It would be nicer to somehow cache the provider but don't think there is a good way in Gradle.

Reply:
did you see this happening in tests? You could temporarily capture and log the stacktrace to figure out what is calling the provider. It definitely seems weird its being invoked multiple times

Reply:
Can verify it with e.g. with [this unit test](https://github.com/palantir/sls-packaging/blob/545a84381ce843cf6a8b34c6c3309725fb906567/gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/pdeps/ResolveProductDependenciesIntegrationSpec.groovy#L50).

These are the stacktraces of the two `DependencyDiscovery#copyConfiguration` invocations. Initial version is the first invocation and the first revision the second invocation: https://gist.github.com/fawind/43830abfe89a33a7ffab85dd22ec4425/revisions?diff=split

Some highlights from the fist invocation stacktrace:
```
org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext
org.gradle.execution.plan.LocalTaskNode.resolveDependencies
org.gradle.initialization.BuildOperationFiringTaskExecutionPreparer$CalculateTaskGraph
```

and from the second invocation stacktrace:
```
org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.lambda$calculateFinalizedValue
org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.visitInputs
org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionState
```

So it looks like the first invocation happens when Gradle builds its task graph and then the second one when the actual task is executed?

Reply:
I think we can instead of the provider use a property with `Property#finalizeValueOnRead` as this will only get resolved once. Pushed the code for this and reverted the changes to `DependencyDiscovery#copyConfiguration`.

Let me add a test overwriting the pdeps configuration though.

Reply:
Hmm, finalizeValueOnRead only exists for Gradle 6.1+ :(

Reply:
we don't need to support older versions anymore, do we?

Reply:
Its probably fine to rely on gradle 6.1 features since it was released over a year and a half ago

Reply:
Agree. Increased the min version to 6.1 and changed the changelog to break to be extra correct.